aten/src/ATen/native/Pooling.cpp

```diff
@@ -127,24 +288,26 @@ Tensor max_pool2d(
     IntArrayRef dilation,
     bool ceil_mode) {
   if (self.is_quantized()) {
-    return at::quantized_max_pool2d(self, kernel_size, stride, padding,
-                                    dilation, ceil_mode);
+    return at::quantized_max_pool2d(
+        self, kernel_size, stride, padding, dilation, ceil_mode);
   }
   if (self.is_mkldnn()) {
     return at::mkldnn_max_pool2d(
         self, kernel_size, stride, padding, dilation, ceil_mode);
   }
-
 #if defined(C10_MOBILE)
-  if(xnnpack::use_max_pool2d(self, kernel_size, padding, stride,
-                             dilation, ceil_mode)) {
+  if (xnnpack::use_max_pool2d(
+          self, kernel_size, padding, stride, dilation, ceil_mode)) {
     return xnnpack::max_pool2d(
         self, kernel_size, padding, stride, dilation, ceil_mode);
   }
 #endif
-  auto output_and_indices = at::max_pool2d_with_indices(
+  if (self.requires_grad()) {
```

Comment:
I think even require grad, we can still use the new impl, anything blocking here?